src/components/event/eventId/status/EventStatusModal.js

```diff
@@ -0,0 +1,104 @@
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Alert, AlertVariants } from '../../../publiq-ui/Alert';
+import { Modal, ModalVariants } from '../../../publiq-ui/Modal';
+import { RadioButtonGroup } from '../../../publiq-ui/RadioButtonGroup';
+import { Stack } from '../../../publiq-ui/Stack';
+import { Text } from '../../../publiq-ui/Text';
+import { TextAreaWithLabel } from '../../../publiq-ui/TextAreaWithLabel';
+import { getValueFromTheme } from '../../../publiq-ui/theme';
+
+const OfferStatus = {
+  AVAILABLE: 'Available',
+  TEMPORARILY_UNAVAILABLE: 'TemporarilyUnavailable',
+  UNAVAILABLE: 'Unavailable',
+};
+
+const maxLengthReason = 200;
+
+const getValue = getValueFromTheme('statusModal');
+
+const EventStatusModal = ({
+  subEvents,
+  visible,
+  className,
+  changeStatuses,
+  onClose,
+}) => {
+  const { t } = useTranslation();
+
+  const [type, setType] = useState('');
+  const [reason, setReason] = useState('');
+
+  return (
+    <Modal
+      visible={visible}
+      title={t('offerStatus.changeStatus')}
+      variant={ModalVariants.QUESTION}
+      size="xl"
+      className={className}
+      confirmTitle={t('offerStatus.actions.save')}
+      cancelTitle={t('offerStatus.actions.close')}
+      onConfirm={() => changeStatuses()}
+      onClose={onClose}
+      confirmButtonDisabled={reason.length > maxLengthReason}
+    >
+      <Stack padding={4} spacing={5}>
```

Comment:
Can this become a separate component? So that it can be reused for places and events?

Reply:
In places this won't be used, because there it is a seperate page and not a modal

Reply:
But the form inside the modal is the same as the form we show on the page for places, no?

Reply:
Yes that's true, apart from the buttons, that live in the modal itself here and are part of the page form itself.

But in one of the previous PR's where I started the work on offerStatus I first had 1 component `OfferStatus`. But then the advice was to split them up because:
1) they would possibly be a bit different (now seems to be quite similar for that specific form that's reused)
2) like this we have the path for the component 1 on 1 equal to the path of the page it's used on, whereas if you have a seperate component it can't live under a specific path like `src/components/event/eventId/status/EventStatusModal.js `

But I agree that this creates duplicate code that does the same thing

Reply:
Yep, I remember that PR and the 1 component. I personally didn't see an issue in having 1 component and tried to explain that I think. The argument was that it would be different for events and places but that's not really correct. There's no difference between events and places in this case, only some differences between calendar types. On the one hand there are `single`, `periodic`, and `permanent`. Those can be on events or places, and will all have the form on the page. Then there's `multiple`, which is only on events and will use the form in the modal with the selection table. So instead of differentiating per offer (having a page/component for event, and different a page/component for place), I would advise differentiating per calendar type instead.

Reply:
So to give an example in pseudo-code:

```
switch (offer.calendarType) {
    case `single`:
    case `periodic`:
    case `permanent`:
        return <ComponentWithStatusFormOnPage />

    case `multiple`:
        return <ComponentWithSelectionTableAndStatusFormInModal />
}
```

And both use a shared `StatusForm` either on the page or in the modal (name is just an example)

Reply:
I'll try to split them up by calendar type then, and the components folder will look something like this then:

```
> components
    > status
       > Single
       > Periodic
       > Permanent
       > Multiple
```

Does this mach what you're thinking @bertramakers?

Reply:
Yes but no 😄 `single`, `periodic` and `permanent` will be the same because they have no subEvents so you always set the status on the "top level" of the offer. While for `multiple` there are subEvents, so you need the selection table + modal and set the status on specific subEvents only.

So a better differentiation would probably be `OfferTopStatus` and `OfferSubEventStatus` or something like that now that I think about it in terms of how the code will be stored in directories etc.

Reply:
I'll give it a crack 😄

Reply:
@bertramakers The forms are different though! The translations used for the radiobuttons are different. Also the api calls used are different.

The component will have more logic inside to show the correct information and do the correct api calls (place vs event)

Reply:
Yeah, there will be some differences but there are ways to make components generic of course. For example for the API callbacks, you can use a callback prop like an `onSave()` instead of having the form do the API call. It's hard to make a decision on whether it's best to make a shared component or make separate components without diving into the actual code though. The translations also seem easy to tackle, for example also passing them as props. If there's more separated logic it might be less interesting to have a generic component, but it's hard to say without more concrete information. It will always be a trade-off between trying to fit too many differences into a generic component vs accidentally having 2 different behaviors for events and places where there should be none because it's two separate components and they have slight accidental differences (now or later after a number of changes), causing UX issues.

Reply:
Refactored 👍🏻